tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/caches/ModelValueCache.scala

```diff
@@ -0,0 +1,35 @@
+package at.forsyte.apalache.tla.bmcmt.caches
+
+import at.forsyte.apalache.tla.bmcmt.{Arena, ArenaCell}
+import at.forsyte.apalache.tla.bmcmt.smt.SolverContext
+import at.forsyte.apalache.tla.bmcmt.types.ConstT
+import at.forsyte.apalache.tla.lir.convenience.tla
+import at.forsyte.apalache.tla.lir.UntypedPredefs._
+import at.forsyte.apalache.tla.typecheck.ModelValueHandler
+
+/**
+ * A cache for model values that are translated as uninterpreted constants, with a unique sort per uniterpreted type.
+ * Since two values are equal iff they are literally the same string, we force
+ * inequality between all the respective SMT constants.
+ *
+ * @author Jure Kukovec
+ */
+class ModelValueCache(solverContext: SolverContext)
+    extends AbstractCache[Arena, (String, String), ArenaCell] with Serializable {
```

Comment:
Why do we need both `ModelValueCache` and `StrValueCache`? For strings, we can use `ModelValueCache` with the type `Str`. There will be no collision on type names, so this should work. We would have to maintain one class instead of two.

Reply:
Yeah, that makes sense. I started out with a different idea for the model cache, but I guess in the latest iteration it can basically replace strings, since they are just 1 standard "uninterpreted type"